db/warehouse/migrate/20240430045111_custom_service_type_slug.rb

```diff
@@ -0,0 +1,28 @@
+class CustomServiceTypeSlug < ActiveRecord::Migration[6.1]
+  def up
+    safety_assured do
+      # change_column :Services, :id, :bigint
+      change_column_null :Services, :EnrollmentID, false
+      change_column_null :Services, :PersonalID, false
+      execute <<~SQL
+        ALTER TABLE "Services" ADD COLUMN custom_service_type_slug VARCHAR GENERATED ALWAYS AS (data_source_id::text || ':' || "TypeProvided"::text || ':' || "RecordType"::text) STORED
+      SQL
+      execute <<~SQL
+        ALTER TABLE "CustomServiceTypes" ADD COLUMN slug VARCHAR GENERATED ALWAYS AS (data_source_id::text || ':' || "hud_type_provided"::text ||':'|| "hud_record_type"::text) STORED
+      SQL
+      execute %{CREATE INDEX idx_custom_service_type_slug ON "CustomServiceTypes" (slug)}
```

Comment:
could add an index on services.custom_service_type_slug but I'm not sure it's worth it, service type seems low cardinality and local perf seems okay without it at least.